website/client/libs/store/helpers/filterTasks.js

```diff
@@ -0,0 +1,83 @@
+import { shouldDo } from 'common/script/cron';
+
+import isEmpty from 'lodash/isEmpty';
+
+// Task filter data
+// @TODO find a way to include user preferences w.r.t sort and defaults
+const taskFilters = {
+  habit: {
+    label: 'habits',
+    filters: [
+      { label: 'all', filterFn: () => true, default: true },
+      { label: 'yellowred', filterFn: t => t.value < 1 }, // weak
+      { label: 'greenblue', filterFn: t => t.value >= 1 }, // strong
+    ],
+  },
+  daily: {
+    label: 'dailies',
+    filters: [
+      { label: 'all', filterFn: () => true, default: true },
+      { label: 'due', filterFn: userPrefs => t => !t.completed && shouldDo(new Date(), t, userPrefs) },
+      { label: 'notDue', filterFn: userPrefs => t => t.completed || !shouldDo(new Date(), t, userPrefs) },
+    ],
+  },
+  todo: {
+    label: 'todos',
+    filters: [
+      { label: 'remaining', filterFn: t => !t.completed, default: true }, // active
+      { label: 'scheduled', filterFn: t => !t.completed && t.date, sort: t => t.date },
+      { label: 'complete2', filterFn: t => t.completed },
+    ],
+  },
+  reward: {
+    label: 'rewards',
+    filters: [
+      { label: 'all', filterFn: () => true, default: true },
+      { label: 'custom', filterFn: () => true }, // all rewards made by the user
+      { label: 'wishlist', filterFn: () => false }, // not user tasks
+    ],
+  },
+};
+
+function typeLabel (filterList) {
+  return (type) => filterList[type].label;
+}
+
+const getTypeLabel = typeLabel(taskFilters);
+
+function filterLabel (filterList) {
+  return (type) => {
+    let filterListByType = filterList[type].filters;
+    let filterListOfLabels = new Array(filterListByType.length);
+    filterListByType.forEach(({ label }, i) => filterListOfLabels[i] = label);
+
+    return filterListOfLabels;
+  };
+}
+
+const getFilterLabels = filterLabel(taskFilters);
+
+function activeFilter (filterList) {
+  return (type, filterType = '') => {
+    let filterListByType = filterList[type].filters;
+    if (isEmpty(filterType)) {
+      return filterListByType.find(f => f.default === true);
+    } else {
+      // check if filter type is available, else send default filter
```

Comment:
is this necessary? I mean, we control which filters are activable it should never happen for the user to select a filter that's unavaible

Reply:
activeFilter (and by extention getActiveFilter) takes two params. param 1 is the task type, while the second is optional filter type (e.g. all, due, notDue for dailies, etc.). I'll simplify the structure and remove the unit test checks for boundary conditions for unavailable filters.